src/eigen.jl

```diff
@@ -124,8 +129,8 @@ end
 end
 
 @inline function _eig(s::Size, A::Base.LinAlg.RealHermSymComplexHerm{T}, permute, scale) where {T <: Real}
-    eigen = eigfact(Hermitian(Array(parent(A))); permute=permute, scale=scale)
-    return (s(eigen.values), s(eigen.vectors)) # Return a SizedArray
+    eigen = eigfact(Hermitian(Array(parent(A))))
+    return (SVector{s[1], T}(eigen.values), SMatrix{s[1], s[2], T}(eigen.vectors)) # Return a SizedArray
```

Comment:
It was intentional that this returned an `SizedArray` rather than a `SArray`.

I'm not saying it was defensible - but it served as an easy way to see that an `Array` was constructed during the process. If we don't want to keep this, then a third possibility is to use `similar_type` to get the output type.